src/typedKnex.ts

```diff
@@ -379,6 +382,40 @@ interface IJoinTableMultipleOnClauses<Model, _SelectableModel, Row> {
     >;
 }
 
+interface IOuterJoinTableMultipleOnClauses<Model, _SelectableModel, Row> {
+    <
+        NewPropertyType,
+        NewPropertyKey extends keyof any
+        >(
+        newPropertyKey: NewPropertyKey,
+        newPropertyClass: new () => NewPropertyType,
+        on: (
+            join: IJoinOnClause2<
+                AddPropertyWithType<Model, NewPropertyKey, NewPropertyType>,
+                NewPropertyType
+            >
+        ) => void
+    ): ITypedQueryBuilder<
+        Model,
```

Comment:
On this and on other joins, setting the type here as Model means that any other previously joined table information is lost until you complete the query. i.e.  if i join table1 with property name t1 and then join table2 with property name t2 and then i start selecting fields I can only access properties from the original table or table2. If I wanted to select from table1 fields I would have to join on table1, select its fields, then join on table2 and select its fields. if you set the return type here to `ITypedQueryBuilder<AddPropertyWithType<Model, NewPropertyKey, NewPropertyType | null>, AddPropertyWithType<Model, NewPropertyKey, NewPropertyType | null>, Row>` you will be able to join two tables in a row and then in a a select statement you can select columns from either table.